src/util/mpir_pmix.inc

```diff
@@ -52,7 +52,14 @@ static int pmix_init(int *has_parent, int *rank, int *size, int *appnum)
     if (pmix_init_count == 1) {
         pmi_errno = PMIx_Init(&pmix_proc, NULL, 0);
         if (pmi_errno == PMIX_ERR_UNREACH) {
-            /* no pmi server, assume we are a singleton */
+            /* failed to connect to server. throw an error if hydra is detected
+             * otherwise proceed as a singleton. */
+            const char *hostname;
+            if (MPL_env2str("PMI_HOSTNAME", &hostname)) {
```

Comment:
Maybe also check for `PMI_FD` or `PMI_PORT`,  one of them must exist for PMI-1 wire protocol.

Reply:
added both